soc/riscv/riscv-ite/it8xxx2/__arithmetic.S

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2022 ITE Corporation.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * When the 'M' extension is disabled, compiler can not recognize div/mul
+ * instructions. So mul/div instructions in the below integer arithmetic
+ * routines are hard coded by opcodes.
+ */
```

Comment:
Ok, let me check IIUC: with M extension disabled, the compiler will call into some library functions to perform multiplication/division. Then, you provide an implementation for those functions which uses M instructions, but add the two instructions needed after them to avoid triggering the bug?

Reply:
You are right.

Reply:
Cool! Maybe make it clear that those two instructions are needed, so someone in the future doesn't try to optimise it and remove then?

Reply:
Will make a follow-up PR to make it clear.